common/error/error.rs

```diff
@@ -219,3 +219,59 @@ macro_rules! typedb_error {
         None
     };
 }
+
```

Comment:
General strategy:
We fail translation with a custom Error (with a variant name `UnimplementedFeature`, or something similar).
Any place which is unreachable thanks to this error, we use a `unimplemented_feature!` macro.